parser/src/main/scala/hmda/parser/fi/FIDataCsvParser.scala

```diff
@@ -13,8 +13,9 @@ class FIDataCsvParser extends FIDataParser[String] {
   def parseLines(lines: Iterable[String]): FIData = {
     val tsLine = lines.head
     val ts = TsCsvParser(tsLine)
-    val lars = lines.tail.map(l => LarCsvParser(l))
-    FIData(ts, lars)
+    // Only parse LARs that do not contain errors
+    val larsWithoutErrors = lines.tail.map(l => LarCsvParser(l)).filter(_.isRight).map(_.right.get)
+    FIData(ts, larsWithoutErrors)
```

Comment:
@kgudel is right. This should return the list of errors when there are errors. As I understand it, we only process a file if it parses 100% correctly. If any lines don't parse, the whole file is invalid and the user must upload a whole new one, so it isn't useful to only return valid lines.

This class isn't being used yet because we haven't implemented the code that parses and validates a whole uploaded file. (Which I'm writing an issue about, and will bring up at standup).